state/factory/workingset.go

```diff
@@ -120,6 +122,13 @@ func (ws *workingSet) runActions(
 			return nil, errors.Wrap(err, "error when run action")
 		}
 		if receipt != nil {
+			if featureCtx.CorrectTxLogIndex {
+				receipt.UpdateIndex(txIndex, logIndex)
```

Comment:
`logIndex = receipt.UpdateIndex(txIndex, logIndex)`

Reply:
updated in 3rd commit